lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/DecimalStatistics.java

```diff
@@ -23,17 +23,18 @@
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
 import static io.airlift.slice.SizeOf.sizeOf;
+import static java.lang.Math.toIntExact;
 
 public class DecimalStatistics
         implements RangeStatistics<BigDecimal>, Hashable
 {
     // 1 byte to denote if null
     public static final long DECIMAL_VALUE_BYTES_OVERHEAD = Byte.BYTES;
 
-    private static final int INSTANCE_SIZE = ClassLayout.parseClass(DecimalStatistics.class).instanceSize();
+    private static final int INSTANCE_SIZE = toIntExact(ClassLayout.parseClass(DecimalStatistics.class).instanceSize());
     // BigDecimal contains BigInteger and BigInteger contains an integer array.
     // The size of the integer array is not accessible from outside; thus rely on callers to tell how large the size is.
-    private static final long BIG_DECIMAL_INSTANCE_SIZE = ClassLayout.parseClass(BigDecimal.class).instanceSize() + ClassLayout.parseClass(BigInteger.class).instanceSize() + sizeOf(new int[0]);
+    private static final long BIG_DECIMAL_INSTANCE_SIZE = toIntExact(ClassLayout.parseClass(BigDecimal.class).instanceSize() + ClassLayout.parseClass(BigInteger.class).instanceSize() + sizeOf(new int[0]));
```

Comment:
What's the point of toIntExact here if BIG_DECIMAL_INSTANCE_SIZE is a long? Maybe change the type to `int`?